pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -225,8 +225,8 @@ func (m *manager) reconcileState() (success []reconciledContainer, failure []rec
 	for _, pod := range m.activePods() {
 		allContainers := pod.Spec.InitContainers
 		allContainers = append(allContainers, pod.Spec.Containers...)
+		status, ok := m.podStatusProvider.GetPodStatus(pod.UID)
```

Comment:
Can we be confident that the `podStatus` will remain consistent over this entire loop? If we can't, I think we open ourselves up to a possible race condition if we check the podStatus at the start of the loop, and then it somehow changes during the middle of executing the loop. Although to be fair, there's also that risk even if we check the pod status during each iteration of the loop, its just less because we are checking the pod status more frequently.

Reply:
@ConnorDoyle 
Can you comment on this refactoring ?

Thanks

Reply:
Yes, this makes sense to do. the pod status really only needs to be queried once before we loop through the containers.